src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -474,23 +450,27 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
     case Event(message: CallStartMessage, data) if !data.isPending(message.callKey) =>
       val updatedData = message.handleStatusPersist(this, data)
       val callKey = message.callKey
-      fetchLocallyQualifiedInputs(callKey) match {
-        case Success(callInputs) =>
-          // TODO: The block of code is only run on non-shards because of issues with DSDEEPB-2490
-          if (!callKey.index.isShard) {
-            val updateDbCallInputs = globalDataAccess.updateCallInputs(workflow.id, callKey, callInputs)
-            updateDbCallInputs onComplete {
-              case Success(i) =>
-                logger.debug(s"$i call input expression(s) updated in database.")
+      callKey match {
```

Comment:
Can this please be pulled out to a function?  IMHO this is too much for a `case`